src/utils/locale.ts

```diff
@@ -61,57 +62,44 @@ function createLocaleNumberFormatter(locale: string): Intl.NumberFormat {
 
 export function delocalizeNumberString(numberString: string, locale: string): string {
   return sanitizeExponentialNumberString(numberString, (nonExpoNumString: string): string => {
-    if (nonExpoNumString) {
-      const groupSeparator = getGroupSeparator(locale);
-      const decimalSeparator = getDecimalSeparator(locale);
-      const minusSign = getMinusSign(locale);
-
-      const splitNumberString = nonExpoNumString.split("");
-      const decimalIndex = splitNumberString.lastIndexOf(decimalSeparator);
-
-      const delocalizedNumberString = splitNumberString
-        .map((value, index) => {
-          if (value === groupSeparator || (value === decimalSeparator && index !== decimalIndex)) {
-            return "";
-          }
-          return value;
-        })
-        .reduce((string, part) => string + part)
-        .replace(decimalSeparator, ".")
-        .replace(minusSign, "-")
-        .replace(/[^0-9\-\.]/g, ""); // remove everything except numbers, "-", and "."
+    const allDecimalsExceptLast = new RegExp(`[.](?=.*[.])`, "g");
+    const everythingExceptNumbersDecimalsAndMinusSigns = new RegExp("[^0-9-.]", "g");
```

Comment:
Can you move this and the above regex outside of the function to be module level? They don't seem to be dynamic, so they can be reused instead of recreating on each call.